### PR TITLE
MTM-57041_Adding information on header values

### PR DESCRIPTION
**Pull request by tsvet**
This information is in the context of https://cumulocity.atlassian.net/browse/MTM-57041 to add information on the not-allowed space in the tenant options, added as header to the request by the microservice management proxy.

### CODE REVIEW

**Review by rpr-github — APPROVED** [review]

**Review by jochenendres — APPROVED** [review]

File: content/microservice-sdk/general-aspects-bundle/security.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Microservices typically provide a REST API and {{< product-c8y-iot >}} provides
 * Authorization: All calls are authenticated using {{< product-c8y-iot >}} credentials with basic or OAuth authorization.
 * TLS Termination: TLS inbound calls are terminated and only HTTP is used inside the cluster.
 * Metering: The API calls are metered in the API calls tenant statistics.
-* Routing: The API gateway routes requests for <kbd>/service/&lt;name&gt;</kbd> to the microservice _&lt;name&gt;_. The request routed to the microservice container and tenant options are added to the request headers. If `contextPath` is defined in the application manifest, the API gateway routes requests for <kbd>/service/&lt;contextPath&gt;</kbd>.
+* Routing: The API gateway routes requests for <kbd>/service/&lt;name&gt;</kbd> to the microservice _&lt;name&gt;_. The request routed to the microservice container and tenant options are added to the request headers. Please note that therefore a space (i.e., " ") is not allowed as a tenant option value.  If `contextPath` is defined in the application manifest, the API gateway routes requests for <kbd>/service/&lt;contextPath&gt;</kbd>.
```
Comment (BeateRixen) [inline content/microservice-sdk/general-aspects-bundle/security.md:12]:
```suggestion
* Routing: The API gateway routes requests for <kbd>/service/&lt;name&gt;</kbd> to the microservice _&lt;name&gt;_. The request routed to the microservice container and tenant options are added to the request headers. Note that therefore a space (i.e., " ") is not allowed as a tenant option value. If `contextPath` is defined in the application manifest, the API gateway routes requests for <kbd>/service/&lt;contextPath&gt;</kbd>.
```

Comment (BeateRixen) [reply content/microservice-sdk/general-aspects-bundle/security.md:12]: I would actaully remove the "i.e." from the brackets and just have (" ") instead.

File: content/microservice-sdk/general-aspects-bundle/security.md [hunk 2]
```diff
@@ -206,6 +206,6 @@ Steps:
...
 
-If a tenant option is created with a key name that starts with "credentials.", it is automatically encrypted and can be fetched as unencrypted only by system users. For instance, when you create a tenant option in a category that matches to the application context path, the value is passed to the microservice by the microservice proxy on the platform as a header (key => value). All encrypted options are decrypted and passed. Moreover, the options can be fetched via REST using the options endpoint at microservice runtime.
+If a tenant option is created with a key name that starts with "credentials.", it is automatically encrypted and can be fetched as unencrypted only by system users. For instance, when you create a tenant option in a category that matches to the application context path, the value is passed to the microservice by the microservice proxy on the platform as a header (key => value). Please note that therefore a space (i.e., " ") is not allowed as a tenant option value. All encrypted options are decrypted and passed. Moreover, the options can be fetched via REST using the options endpoint at microservice runtime.
```
Comment (BeateRixen) [inline content/microservice-sdk/general-aspects-bundle/security.md:209]:
```suggestion
If a tenant option is created with a key name that starts with "credentials.", it is automatically encrypted and can be fetched as unencrypted only by system users. For instance, when you create a tenant option in a category that matches to the application context path, the value is passed to the microservice by the microservice proxy on the platform as a header (key => value). Note that therefore a space (i.e., " ") is not allowed as a tenant option value. All encrypted options are decrypted and passed. Moreover, the options can be fetched via REST using the options endpoint at microservice runtime.
```

Comment (BeateRixen) [reply content/microservice-sdk/general-aspects-bundle/security.md:209]: See above. I would remove the "i.e." from the brackets and just have (" ") instead.

File: content/microservice-sdk/general-aspects-bundle/security.md (hunk 1 above)
Comment (tsvet) [reply content/microservice-sdk/general-aspects-bundle/security.md]: Please see 06cfc3d7a463f13fd97eadea6dafcd0227dc2f02

File: content/microservice-sdk/general-aspects-bundle/security.md (hunk 2 above)
Comment (tsvet) [reply content/microservice-sdk/general-aspects-bundle/security.md]: Please see 06cfc3d7a463f13fd97eadea6dafcd0227dc2f02

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**